### DISCUSSION TITLE
Good training resources available for Azure newbies ?

### DISCUSSION THREAD

**Discussion by stephenhauck** [Azure/azure-mobile-apps#473] [Q&A]
I did the ToDo demo and it;'s VERY interesting 
I added the authentication and that's interesting as well .. 

I have questions ... 
- Can I run SQLite on the backend and on the clients or do I need SQL Server ?
- How do I implement things in such a way that when the use logs in they are associated with a customer so I can run SQL Server and segment the data by customer or is there a better way to do that ?
- Is there an event somewhere that shows the synchronization progress so the user can be alerted to "it's working" and "it's done" ?

Sorry but I used to writing sync manually and having lots of control and if I don't need to that's cool I just need to know how it works a little better

**Comment by adrianhall** [comment]
Fairly straight forward:

1. Don't use SQLite on services ever in production.   It's not a suitable database.  You can, however, use any other database supported by Entity Framework Core.  I've used Cosmos, SQL Server, MySQL, and PostgreSQL to good effect.
2. You need to create a custom repository for customer segmentation.  It's not built in.   See https://adrianhall.github.io/azure/2021/11/12/zumo-irepository/
3. Not right now, but there is an issue for it.  It will come in the next couple of releases.

**Reply by stephenhauck** [reply]
With regards to item 2 ... in the server code the TableController<T> takes care of the CRUD for the ToDoItem
I need to know specifically how to integrate the customer id in that case by default or do I have to then leave the TableController and build out the code manually ?

I mean there's no code for the CRUD and that's awesome but I need to associate the CRUD operations with a specific customer or user or both to segment the data ...

**Reply by adrianhall** [reply]
There are two ways of getting where you need to be.  The best way is to create your own IRepository<T> implementation that does the necessary logic to associate the user ID with the customer ID and only return the data that is required.   The article referenced tells you how to write an IRepository<T> implementation.
